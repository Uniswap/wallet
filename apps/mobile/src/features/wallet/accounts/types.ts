import { Palette } from 'src/styles/color'

export enum AccountType {
  SignerMnemonic = 'signerMnemonic', // Key lives in native keystore
  Readonly = 'readonly', // Accounts without keys (e.g. so user can track balances)
}

export enum BackupType {
  Manual = 'manual',
  Cloud = 'cloud',
}

type DynamicPalette = Pick<Palette, 'userThemeColor'>

export type AccountCustomizations = {
  palette?: DynamicPalette
  localPfp?: string
}

export interface AccountBase {
  type: AccountType
  address: Address
  name?: string
  customizations?: AccountCustomizations
  backups?: BackupType[]
  flashbotsEnabled?: boolean
  pending?: boolean
  timeImportedMs: number
  pushNotificationsEnabled?: boolean
  showSmallBalances?: boolean
  showSpamTokens?: boolean
}

export interface SignerMnemonicAccount extends AccountBase {
  type: AccountType.SignerMnemonic
  derivationIndex: number
  mnemonicId: string
}

export interface ReadOnlyAccount extends AccountBase {
  type: AccountType.Readonly
}

export type Account = SignerMnemonicAccount | ReadOnlyAccount
