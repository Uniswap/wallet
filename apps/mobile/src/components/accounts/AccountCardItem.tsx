import React, { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import ContextMenu from 'react-native-context-menu-view'
import { useAppDispatch, useAppTheme } from 'src/app/hooks'
import { navigate } from 'src/app/navigation/rootNavigation'
import Check from 'src/assets/icons/check.svg'
import { AccountIcon } from 'src/components/AccountIcon'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Flex } from 'src/components/layout'
import { NotificationBadge } from 'src/components/notifications/Badge'
import { Text } from 'src/components/Text'
import { useENSAvatar } from 'src/features/ens/api'
import { closeModal } from 'src/features/modals/modalSlice'
import { useSelectAddressHasNotifications } from 'src/features/notifications/hooks'
import { pushNotification } from 'src/features/notifications/notificationSlice'
import { AppNotificationType, CopyNotificationType } from 'src/features/notifications/types'
import { ModalName } from 'src/features/telemetry/constants'
import { useDisplayName } from 'src/features/wallet/hooks'
import { Screens } from 'src/screens/Screens'
import { iconSizes } from 'src/styles/sizing'
import { setClipboard } from 'src/utils/clipboard'
import { formatUSDPrice, NumberType } from 'src/utils/format'

type AccountCardItemProps = {
  address: Address
  isActive?: boolean
  isViewOnly: boolean
  onPress: (address: Address) => void
} & PortfolioValueProps

type PortfolioValueProps = {
  isPortfolioValueLoading: boolean
  portfolioValue: number | undefined
}

function PortfolioValue({
  isPortfolioValueLoading,
  portfolioValue,
}: PortfolioValueProps): JSX.Element {
  const isLoading = isPortfolioValueLoading && portfolioValue === undefined

  return (
    <Text
      color="textTertiary"
      loading={isLoading}
      loadingPlaceholderText="$000.00"
      variant="subheadSmall">
      {formatUSDPrice(portfolioValue, NumberType.PortfolioBalance)}
    </Text>
  )
}

export function AccountCardItem({
  address,
  isViewOnly,
  isActive,
  isPortfolioValueLoading,
  portfolioValue,
  onPress,
}: AccountCardItemProps): JSX.Element {
  const { t } = useTranslation()

  const theme = useAppTheme()
  const displayName = useDisplayName(address)
  const { data: avatar } = useENSAvatar(address)
  const hasNotifications = useSelectAddressHasNotifications(address)
  const dispatch = useAppDispatch()

  const icon = useMemo(() => {
    return (
      <AccountIcon
        address={address}
        avatarUri={avatar}
        showViewOnlyBadge={isViewOnly}
        size={iconSizes.icon40}
      />
    )
  }, [address, avatar, isViewOnly])

  const onPressCopyAddress = (): void => {
    setClipboard(address)
    dispatch(
      pushNotification({
        type: AppNotificationType.Copied,
        copyType: CopyNotificationType.Address,
      })
    )
  }

  const onPressWalletSettings = (): void => {
    dispatch(closeModal({ name: ModalName.AccountSwitcher }))
    navigate(Screens.SettingsStack, {
      screen: Screens.SettingsWallet,
      params: { address },
    })
  }

  const menuActions = useMemo(() => {
    return [
      { title: t('Copy wallet address'), systemIcon: 'doc.on.doc' },
      { title: t('Wallet settings'), systemIcon: 'gearshape' },
    ]
  }, [t])

  return (
    <ContextMenu
      actions={menuActions}
      onPress={(e): void => {
        // Emitted index based on order of menu action array
        // Copy address
        if (e.nativeEvent.index === 0) {
          onPressCopyAddress()
        }
        // Navigate to settings
        if (e.nativeEvent.index === 1) {
          onPressWalletSettings()
        }
      }}>
      <TouchableArea
        hapticFeedback
        pb="spacing12"
        pt="spacing8"
        px="spacing24"
        onPress={(): void => onPress(address)}>
        <Flex row alignItems="center" testID={`account_item/${address}`}>
          <Flex row shrink alignItems="center">
            <NotificationBadge showIndicator={hasNotifications}>{icon}</NotificationBadge>
            <Flex fill gap="none">
              <Text numberOfLines={1} variant="subheadLarge">
                {displayName?.name}
              </Text>
              <PortfolioValue
                isPortfolioValueLoading={isPortfolioValueLoading}
                portfolioValue={portfolioValue}
              />
            </Flex>
          </Flex>
          <Flex row alignItems="center" gap="none">
            {isActive && (
              <Check
                color={theme.colors.userThemeMagenta}
                height={theme.iconSizes.icon24}
                width={theme.iconSizes.icon24}
              />
            )}
          </Flex>
        </Flex>
      </TouchableArea>
    </ContextMenu>
  )
}
