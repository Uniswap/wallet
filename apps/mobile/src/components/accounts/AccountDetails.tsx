import React from 'react'
import { useAppTheme } from 'src/app/hooks'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { Chevron } from 'src/components/icons/Chevron'
import { Flex } from 'src/components/layout'
import { Text } from 'src/components/Text'
import { shortenAddress } from 'src/utils/addresses'

export function AccountDetails({
  address,
  iconSize = 20,
  chevron = false,
}: {
  address: string
  iconSize?: number
  chevron?: boolean
}): JSX.Element {
  const theme = useAppTheme()

  return (
    <Flex row shrink alignItems="center" justifyContent="space-between">
      <Flex row shrink flex={1}>
        <AddressDisplay
          hideAddressInSubtitle
          address={address}
          horizontalGap="spacing8"
          size={iconSize}
          variant="subheadSmall"
        />
      </Flex>
      <Flex row shrink flex={1} gap="spacing4" justifyContent="flex-end">
        <Text color="textSecondary" variant="subheadSmall">
          {shortenAddress(address)}
        </Text>
        {chevron && (
          <Chevron
            color={theme.colors.textSecondary}
            direction="e"
            height={iconSize}
            width={iconSize}
          />
        )}
      </Flex>
    </Flex>
  )
}
