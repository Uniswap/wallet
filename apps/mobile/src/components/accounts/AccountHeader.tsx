import { impactAsync, ImpactFeedbackStyle, selectionAsync } from 'expo-haptics'
import React, { useCallback } from 'react'
import { useAppDispatch, useAppSelector, useAppTheme } from 'src/app/hooks'
import { navigate } from 'src/app/navigation/rootNavigation'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Chevron } from 'src/components/icons/Chevron'
import { Flex } from 'src/components/layout'
import { Box } from 'src/components/layout/Box'
import { openModal } from 'src/features/modals/modalSlice'
import { ElementName, ModalName } from 'src/features/telemetry/constants'
import { Screens } from 'src/screens/Screens'
import { setClipboard } from 'src/utils/clipboard'
import { isDevBuild } from 'src/utils/version'
import SettingsIcon from 'ui/src/assets/icons/settings.svg'
import { iconSizes } from 'ui/src/theme/iconSizes'
import { pushNotification } from 'wallet/src/features/notifications/slice'
import { AppNotificationType, CopyNotificationType } from 'wallet/src/features/notifications/types'
import { selectActiveAccountAddress } from 'wallet/src/features/wallet/selectors'

export function AccountHeader(): JSX.Element {
  const theme = useAppTheme()
  const activeAddress = useAppSelector(selectActiveAccountAddress)
  const dispatch = useAppDispatch()

  const onPressAccountHeader = useCallback(() => {
    dispatch(openModal({ name: ModalName.AccountSwitcher }))
  }, [dispatch])

  const onPressSettings = (): void => {
    navigate(Screens.SettingsStack, { screen: Screens.Settings })
  }

  const onPressCopyAddress = async (): Promise<void> => {
    if (activeAddress) {
      await impactAsync()
      await setClipboard(activeAddress)
      dispatch(
        pushNotification({
          type: AppNotificationType.Copied,
          copyType: CopyNotificationType.Address,
        })
      )
    }
  }

  return (
    <Box
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
      mt="spacing16"
      testID="account-header">
      <TouchableArea
        hapticFeedback
        alignItems="center"
        flex={1}
        flexDirection="row"
        hapticStyle={ImpactFeedbackStyle.Medium}
        mr="spacing12"
        testID={ElementName.Manage}
        onLongPress={async (): Promise<void> => {
          await onPressCopyAddress()
          if (isDevBuild()) {
            await selectionAsync()
            dispatch(openModal({ name: ModalName.Experiments }))
          }
        }}
        onPress={onPressAccountHeader}>
        {activeAddress && (
          <Flex row alignItems="center" gap="spacing4">
            <Flex shrink>
              <AddressDisplay
                hideAddressInSubtitle
                address={activeAddress}
                horizontalGap="spacing8"
                size={iconSizes.icon28}
                variant="subheadLarge"
              />
            </Flex>
            <Chevron
              color={theme.colors.textTertiary}
              direction="s"
              height={iconSizes.icon20}
              width={iconSizes.icon20}
            />
          </Flex>
        )}
      </TouchableArea>
      <TouchableArea hapticFeedback onPress={onPressSettings}>
        <SettingsIcon
          color={theme.colors.textTertiary}
          height={theme.iconSizes.icon28}
          opacity="0.8"
          width={theme.iconSizes.icon28}
        />
      </TouchableArea>
    </Box>
  )
}
