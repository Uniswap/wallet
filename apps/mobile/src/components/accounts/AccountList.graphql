query AccountList($addresses: [String!]!) {
  portfolios(ownerAddresses: $addresses) {
    id
    tokensTotalDenominatedValue {
      id
      value
    }
  }
}