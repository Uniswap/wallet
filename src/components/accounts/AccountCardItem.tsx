import React, { useMemo } from 'react'
import { useAppTheme } from 'src/app/hooks'
import Check from 'src/assets/icons/check.svg'
import TripleDots from 'src/assets/icons/triple-dots.svg'
import { AccountIcon } from 'src/components/AccountIcon'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Flex } from 'src/components/layout'
import { NotificationBadge } from 'src/components/notifications/Badge'
import { Text } from 'src/components/Text'
import { DecimalNumber } from 'src/components/text/DecimalNumber'
import { useENSAvatar } from 'src/features/ens/api'
import { useSelectAddressHasNotifications } from 'src/features/notifications/hooks'
import { ElementName } from 'src/features/telemetry/constants'
import { useDisplayName } from 'src/features/wallet/hooks'
import { iconSizes } from 'src/styles/sizing'
import { formatUSDPrice, NumberType } from 'src/utils/format'

type AccountCardItemProps = {
  address: Address
  isActive?: boolean
  isViewOnly: boolean
  onPress: (address: Address) => void
  onPressEdit: (address: Address) => void
} & PortfolioValueProps

type PortfolioValueProps = {
  isPortfolioValueLoading: boolean
  portfolioValue: number | undefined
}

function PortfolioValue({
  isPortfolioValueLoading,
  portfolioValue,
}: PortfolioValueProps): JSX.Element {
  const isLoading = isPortfolioValueLoading && portfolioValue === undefined

  return (
    <DecimalNumber
      color="textSecondary"
      loading={isLoading}
      number={formatUSDPrice(portfolioValue, NumberType.PortfolioBalance)}
      variant="bodySmall"
    />
  )
}

export function AccountCardItem({
  address,
  isViewOnly,
  isActive,
  isPortfolioValueLoading,
  portfolioValue,
  onPress,
  onPressEdit,
}: AccountCardItemProps): JSX.Element {
  const theme = useAppTheme()
  const displayName = useDisplayName(address)
  const { data: avatar } = useENSAvatar(address)
  const hasNotifications = useSelectAddressHasNotifications(address)

  const icon = useMemo(() => {
    return (
      <AccountIcon
        address={address}
        avatarUri={avatar}
        showViewOnlyBadge={isViewOnly}
        size={iconSizes.icon40}
      />
    )
  }, [address, avatar, isViewOnly])

  return (
    <TouchableArea
      hapticFeedback
      pb="spacing12"
      pt="spacing8"
      px="spacing24"
      onPress={(): void => onPress(address)}>
      <Flex row alignItems="center" testID={`account_item/${address}`}>
        <Flex row shrink alignItems="center">
          <NotificationBadge showIndicator={hasNotifications}>{icon}</NotificationBadge>
          <Flex fill gap="none">
            <Text numberOfLines={1} variant="bodyLarge">
              {displayName?.name}
            </Text>
            <PortfolioValue
              isPortfolioValueLoading={isPortfolioValueLoading}
              portfolioValue={portfolioValue}
            />
          </Flex>
        </Flex>
        <Flex row alignItems="center" gap="none">
          {isActive && (
            <Check
              color={theme.colors.userThemeMagenta}
              height={theme.iconSizes.icon20}
              width={theme.iconSizes.icon20}
            />
          )}
          <TouchableArea
            name={ElementName.Edit}
            pl="spacing12"
            py="spacing16"
            onPress={(): void => onPressEdit(address)}>
            <TripleDots
              color={theme.colors.textTertiary}
              height={iconSizes.icon12}
              strokeLinecap="round"
              strokeWidth="1"
              width={iconSizes.icon16}
            />
          </TouchableArea>
        </Flex>
      </Flex>
    </TouchableArea>
  )
}
