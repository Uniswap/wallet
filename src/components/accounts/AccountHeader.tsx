import { ImpactFeedbackStyle, selectionAsync } from 'expo-haptics'
import React, { useCallback } from 'react'
import { useAppDispatch, useAppSelector, useAppTheme } from 'src/app/hooks'
import { navigate } from 'src/app/navigation/rootNavigation'
import DoubleChevron from 'src/assets/icons/double-chevron.svg'
import ScanQRWCIcon from 'src/assets/icons/scan-qr-wc.svg'
import SettingsIcon from 'src/assets/icons/settings.svg'
import { AddressDisplay } from 'src/components/AddressDisplay'
import { TouchableArea } from 'src/components/buttons/TouchableArea'
import { Flex } from 'src/components/layout'
import { Box } from 'src/components/layout/Box'
import { ScannerModalState } from 'src/components/QRCodeScanner/constants'
import { openModal } from 'src/features/modals/modalSlice'
import { ElementName, ModalName } from 'src/features/telemetry/constants'
import { selectActiveAccountAddress } from 'src/features/wallet/selectors'
import { removePendingSession } from 'src/features/walletConnect/walletConnectSlice'
import { Screens } from 'src/screens/Screens'
import { iconSizes } from 'src/styles/sizing'
import { isDevBuild } from 'src/utils/version'
function QRScannerIconButton({ onPress }: { onPress: () => void }): JSX.Element {
  const theme = useAppTheme()

  return (
    <TouchableArea hapticFeedback name={ElementName.WalletConnectScan} onPress={onPress}>
      <ScanQRWCIcon
        color={theme.colors.textSecondary}
        height={theme.iconSizes.icon24}
        strokeWidth={2}
      />
    </TouchableArea>
  )
}

export function AccountHeader(): JSX.Element {
  const theme = useAppTheme()
  const activeAddress = useAppSelector(selectActiveAccountAddress)
  const dispatch = useAppDispatch()

  const onPressAccountHeader = useCallback(() => {
    dispatch(openModal({ name: ModalName.AccountSwitcher }))
  }, [dispatch])

  const onPressSettings = (): void => {
    navigate(Screens.SettingsStack, { screen: Screens.Settings })
  }

  const onPressScan = useCallback(() => {
    // in case we received a pending session from a previous scan after closing modal
    dispatch(removePendingSession())
    dispatch(
      openModal({ name: ModalName.WalletConnectScan, initialState: ScannerModalState.ScanQr })
    )
  }, [dispatch])

  return (
    <Box
      alignItems="center"
      flexDirection="row"
      justifyContent="space-between"
      mt="spacing16"
      testID="account-header">
      <TouchableArea
        hapticFeedback
        alignItems="center"
        flex={1}
        flexDirection="row"
        hapticStyle={ImpactFeedbackStyle.Medium}
        mr="spacing12"
        name={ElementName.Manage}
        testID={ElementName.Manage}
        onLongPress={(): void => {
          if (isDevBuild()) {
            selectionAsync()
            dispatch(openModal({ name: ModalName.Experiments }))
          }
        }}
        onPress={onPressAccountHeader}>
        {activeAddress && (
          <Flex row alignItems="center" gap="spacing4">
            <Flex shrink>
              <AddressDisplay
                hideAddressInSubtitle
                address={activeAddress}
                horizontalGap="spacing8"
                size={iconSizes.icon28}
                variant="subheadLarge"
              />
            </Flex>
            <DoubleChevron
              color={theme.colors.textSecondary}
              height={iconSizes.icon16}
              width={iconSizes.icon16}
            />
          </Flex>
        )}
      </TouchableArea>
      <Flex alignItems="center" flexDirection="row" gap="spacing16" justifyContent="flex-end">
        <QRScannerIconButton onPress={onPressScan} />
        <TouchableArea onPress={onPressSettings}>
          <Flex row alignItems="center">
            <SettingsIcon
              color={theme.colors.textSecondary}
              height={theme.iconSizes.icon24}
              width={theme.iconSizes.icon24}
            />
          </Flex>
        </TouchableArea>
      </Flex>
    </Box>
  )
}
