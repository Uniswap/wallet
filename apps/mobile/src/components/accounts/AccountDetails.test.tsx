import React from 'react'
import { AccountDetails } from 'src/components/accounts/AccountDetails'
import { account } from 'src/test/fixtures'
import { render } from 'src/test/test-utils'

describe(AccountDetails, () => {
  it('renders without error', () => {
    const tree = render(<AccountDetails address={account.address} iconSize={50} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders without error with chevron', () => {
    const tree = render(<AccountDetails chevron address={account.address} iconSize={50} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
