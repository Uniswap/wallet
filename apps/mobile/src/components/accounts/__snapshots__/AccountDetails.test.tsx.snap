// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails renders without error 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 999999,
              "borderWidth": 0,
              "display": "flex",
              "position": "relative",
            },
            Object {
              "padding": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "height": 50,
                "width": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "height": 50,
                  "width": 50,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                clip={true}
                mask={
                  <UniconMask
                    attributeData={
                      Object {
                        "0": "#5CFE9D",
                        "1": "#B8C3B7",
                        "2": Array [
                          Object {
                            "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                          },
                        ],
                        "3": Array [
                          Object {
                            "fillType": "evenOdd",
                            "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                          },
                        ],
                      }
                    }
                    showBorder={false}
                    size={50}
                  />
                }
              >
                <View
                  transform={
                    Array [
                      Object {
                        "scale": 1.3513513513513513,
                      },
                    ]
                  }
                >
                  <View
                    height={37}
                    width={37}
                    x={0}
                    y={0}
                  >
                    <View
                      colors={
                        Array [
                          "#5CFE9D",
                          "#B8C3B7",
                        ]
                      }
                      end={null}
                      start={null}
                    />
                  </View>
                  <View
                    color="#EDE590"
                    cx={18.5}
                    cy={-13.36111111111111}
                    r={30.833333333333336}
                  >
                    <View
                      blur={15}
                    />
                  </View>
                </View>
              </View>
              <View
                clip={true}
                mask={
                  <UniconMask
                    attributeData={
                      Object {
                        "0": "#5CFE9D",
                        "1": "#B8C3B7",
                        "2": Array [
                          Object {
                            "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                          },
                        ],
                        "3": Array [
                          Object {
                            "fillType": "evenOdd",
                            "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                          },
                        ],
                      }
                    }
                    overlay={true}
                    showBorder={false}
                    size={50}
                  />
                }
              >
                <View
                  color="#000000"
                  height={50}
                  opacity={0.08}
                  width={50}
                  x={0}
                  y={0}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "width": 8,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            allowFontScaling={true}
            ellipsizeMode="tail"
            maxFontSizeMultiplier={1.4}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#0E1524",
                  "display": "flex",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 15,
                  "lineHeight": 20,
                },
              ]
            }
            testID="address-display/name/0x​f39F...2266"
          >
            0x​f39F...2266
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        Array [
          Object {
            "color": "#5D6785",
            "display": "flex",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "lineHeight": 20,
          },
        ]
      }
    >
      0xf39F...2266
    </Text>
  </View>
</View>
`;

exports[`AccountDetails renders without error with chevron 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 999999,
              "borderWidth": 0,
              "display": "flex",
              "position": "relative",
            },
            Object {
              "padding": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "height": 50,
                "width": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "height": 50,
                  "width": 50,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                clip={true}
                mask={
                  <UniconMask
                    attributeData={
                      Object {
                        "0": "#5CFE9D",
                        "1": "#B8C3B7",
                        "2": Array [
                          Object {
                            "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                          },
                        ],
                        "3": Array [
                          Object {
                            "fillType": "evenOdd",
                            "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                          },
                        ],
                      }
                    }
                    showBorder={false}
                    size={50}
                  />
                }
              >
                <View
                  transform={
                    Array [
                      Object {
                        "scale": 1.3513513513513513,
                      },
                    ]
                  }
                >
                  <View
                    height={37}
                    width={37}
                    x={0}
                    y={0}
                  >
                    <View
                      colors={
                        Array [
                          "#5CFE9D",
                          "#B8C3B7",
                        ]
                      }
                      end={null}
                      start={null}
                    />
                  </View>
                  <View
                    color="#EDE590"
                    cx={18.5}
                    cy={-13.36111111111111}
                    r={30.833333333333336}
                  >
                    <View
                      blur={15}
                    />
                  </View>
                </View>
              </View>
              <View
                clip={true}
                mask={
                  <UniconMask
                    attributeData={
                      Object {
                        "0": "#5CFE9D",
                        "1": "#B8C3B7",
                        "2": Array [
                          Object {
                            "path": "M34 20C34 28.8366 26.8366 36 18 36C9.16344 36 2 28.8366 2 20L2 -3.8147e-06L34 -1.01717e-06V20Z",
                          },
                        ],
                        "3": Array [
                          Object {
                            "fillType": "evenOdd",
                            "path": "M16 8C12.2937 6.62854 9.37146 3.70632 8 0C6.62854 3.70632 3.70632 6.62854 0 8C3.70632 9.37146 6.62854 12.2937 8 16C9.37146 12.2937 12.2937 9.37146 16 8ZM8 11C9.65685 11 11 9.65685 11 8C11 6.34315 9.65685 5 8 5C6.34315 5 5 6.34315 5 8C5 9.65685 6.34315 11 8 11Z",
                          },
                        ],
                      }
                    }
                    overlay={true}
                    showBorder={false}
                    size={50}
                  />
                }
              >
                <View
                  color="#000000"
                  height={50}
                  opacity={0.08}
                  width={50}
                  x={0}
                  y={0}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "width": 8,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            allowFontScaling={true}
            ellipsizeMode="tail"
            maxFontSizeMultiplier={1.4}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#0E1524",
                  "display": "flex",
                  "fontFamily": "Inter-Medium",
                  "fontSize": 15,
                  "lineHeight": 20,
                },
              ]
            }
            testID="address-display/name/0x​f39F...2266"
          >
            0x​f39F...2266
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 16,
          "width": 16,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        Array [
          Object {
            "color": "#5D6785",
            "display": "flex",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "lineHeight": 20,
          },
        ]
      }
    >
      0xf39F...2266
    </Text>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexGrow": 0,
            "flexShrink": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 999999,
            "display": "flex",
            "justifyContent": "center",
          },
          Object {
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        fill="none"
        focusable={false}
        height={50}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        vbHeight={24}
        vbWidth={26}
        width={50}
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M15 6L9 12L15 18"
            fill={
              Object {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              Object {
                "payload": 4284311429,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="3"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
